frontend/src/components/mine/ContactInfo/ViewPartyRelationships.js

```diff
@@ -162,148 +207,184 @@ export class ViewPartyRelationships extends Component {
     });
   }
 
+  renderMenu = (partyRelationshipGroupingLevels) => (
+    <Menu>
+      {partyRelationshipGroupingLevels.map((group) => [
+        this.props.partyRelationshipTypes
+          .filter((x) => x.grouping_level === group)
+          .map((value) => (
+            <Menu.Item key={value.mine_party_appt_type_code}>
+              <button
+                className="full"
+                type="button"
+                onClick={() => {
+                  this.setState({
+                    selectedPartyRelationshipType: value.mine_party_appt_type_code,
+                  });
+                  this.openAddPartyRelationshipModal(
+                    value,
+                    this.onSubmitAddPartyRelationship,
+                    this.props.handleChange,
+                    this.onPartySubmit,
+                    ModalContent.ADD_CONTACT,
+                    this.props.mine
+                  );
+                }}
+              >
+                {`${value.description}`}
+              </button>
+            </Menu.Item>
+          )),
+        <Menu.Divider key={group} />,
+      ])}
+    </Menu>
+  );
+
   renderPartyRelationship = (partyRelationship) => {
-    if (!this.props.partyRelationshipTypes) return <div />;
+    if (
+      !this.props.partyRelationshipTypesList.length > 0 ||
```

Comment:
It seems to me that this is checking `this.props.partyRelationshipTypesList.length <= 0`? Avoiding the negation seems clearer to me

Reply:
Agree. Changed.